server/src/capture.rs

```diff
@@ -0,0 +1,227 @@
+}
+ */
+pub struct EventCapture {
+    db_client: Arc<Mutex<Client>>,
+}
+
+impl EventCapture {
+    /*
+        Creates a new `EventCapture` instance by reading the database connection parameters
+        from the `config.json` file and connecting to the PostgreSQL database.
+
+        # Arguments
+        - `config_path`: The file path to the `config.json` file.
+
+        # Returns
+            A `Result` containing an `EventCapture` instance or a `Box<dyn std::error::Error>`.
+
+    */
+
+    pub async fn new<P: AsRef<Path>>(config_path: P) -> Result<Self, io::Error> {
+        // Read the configuration file
+        let config_content =
+            fs::read_to_string(config_path).map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+        let config: Config = serde_json::from_str(&config_content)
+            .map_err(|e| io::Error::new(io::ErrorKind::InvalidData, e))?;
+
+        // Build the connection string for the PostgreSQL database
+        let conn_str = format!(
+            "host={} user={} password={} dbname={}",
+            config.db_ip, config.db_user, config.db_password, config.db_name
+        );
+
+        info!(
+            "Attempting Capture Database Connection. IP:[{}] Username:[{}] Database Name:[{}]",
+            config.db_ip, config.db_user, config.db_name
+        );
+
+        // Connect to the database asynchronously
+        let (client, connection) = tokio_postgres::connect(&conn_str, NoTls)
+            .await
+            .map_err(|e| io::Error::new(io::ErrorKind::ConnectionRefused, e))?;
+
+        // Spawn a task to manage the database connection in the background
+        tokio::spawn(async move {
+            if let Err(e) = connection.await {
+                error!("Database connection error: [{}]", e);
+            }
+        });
+
+        info!(
+            "Connected to Database [{}] as User [{}]",
+            config.db_name, config.db_user
+        );
+
+        Ok(EventCapture {
+            db_client: Arc::new(Mutex::new(client)),
+        })
+    }
+
+    /*
+    Inserts an event into the database.
+
+    # Arguments
+    - `event`: An `Event` instance containing the event data to insert.
+
+    # Returns
+    A `Result` indicating success or containing a `tokio_postgres::Error`.
+
+    # Example
+    #[tokio::main]
+    async fn main() -> Result<(), Box<dyn std::error::Error>> {
+        let event_capture = EventCapture::new("config.json").await?;
+
+        let event = Event {
+            user_id: "user123".to_string(),
+            event_type: "keystroke".to_string(),
+            data: Some("Pressed key A".to_string()),
+        };
+
+        event_capture.insert_event(event).await?;
+        Ok(())
+    }
+    */
+    pub async fn insert_event(&self, event: Event) -> Result<(), io::Error> {
+        let current_time = Local::now();
+        let formatted_time = current_time.to_rfc3339();
+
+        // SQL statement to insert the event into the 'events' table
+        let stmt = "
```

Comment:
`use indoc::indoc` then `let stmt = indoc!("stuff")` to dedent text automatically. (Not a big deal, since it's SQL, but good practice.)

Reply:
I will have to read up on this.  I am not familiar with what you are saying here